numpy/core/numerictypes.py

```diff
@@ -210,8 +210,20 @@ def english_capitalize(s):
 
 
 sctypeDict = {}      # Contains all leaf-node scalar types with aliases
-sctypeNA = {}        # Contails all leaf-node types -> numarray type equivalences
-allTypes = {}      # Collect the types we will add to the module here
+class TypeNADict(dict):
+    def __getitem__(self, key):
+        # 2018-06-24, 1.16
+        warnings.warn('sctypeNA and typeNA will be removed in v1.17 '
```

Comment:
Do we want to commit to this fast a deprecation cycle? If so, this needs a VisibleDeprecationWarning probably. Might be worth getting more feedback on that.

Reply:
will add the visible warning and delay to 1.18